src/Workspaces/CoreTest/SymbolKeyTests.cs

```diff
@@ -113,6 +113,62 @@ public class B { };
             TestRoundTrip(GetDeclaredSymbols(compilation), compilation);
         }
 
+        [Fact]
+        public void TestMissingField1()
+        {
+            var source = @"
+
+public class C
+{
+    int;
+}
+";
+            var compilation = GetCompilation(source, LanguageNames.CSharp);
+            TestRoundTrip(GetDeclaredSymbols(compilation), compilation);
+        }
+
+        [Fact]
+        public void TestMissingField2()
+        {
+            var source = @"
+
+public class C
+{
+    int a,;
+}
+";
+            var compilation = GetCompilation(source, LanguageNames.CSharp);
+            TestRoundTrip(GetDeclaredSymbols(compilation), compilation);
+        }
+
+        [Fact]
+        public void TestMissingEvent1()
+        {
+            var source = @"
+
+public class C
+{
+    event System.Action;
+}
+";
+            var compilation = GetCompilation(source, LanguageNames.CSharp);
+            TestRoundTrip(GetDeclaredSymbols(compilation), compilation);
+        }
+
+        [Fact]
+        public void TestMissingEvent2()
+        {
+            var source = @"
+
+public class C
+{
+    event System.Action a,;
+}
+";
+            var compilation = GetCompilation(source, LanguageNames.CSharp);
+            TestRoundTrip(GetDeclaredSymbols(compilation), compilation);
+        }
```

Comment:
Based on the test names, it looks like `GetDeclaredSymbols(compilation)` doesn't return the symbol from erroneous declaration? Is that correct? Maybe consider assert on returned symbols like the test below to make it clear?

Reply:
ah.  it does actually return them.  but an asseertion makes sense.  wil. add.